FBKVOController/FBKVOController.m

```diff
@@ -275,12 +288,20 @@ - (void)unobserve:(id)object info:(_FBKVOInfo *)info
   if (nil == info) {
     return;
   }
-  
+
+  if (!info->_registered) {
+      // this could happen when `NSKeyValueObservingOptionInitial` is one of the NSKeyValueObservingOptions.
+      // we cannot unobserve them right now because they are not technially observers yet.
+      // instead, we set a flag so they can be unobserved later, after the `addObserver` call returns.
+      info->_unregisteredBeforeRegistered = YES;
+      return;
```

Comment:
Sounds good. You are right, nevermind those comments - since you are calling it twice the first pass will remove it, the first will be a no-op and the second call will remove the observation. For some reason I constantly make mistakes in basic boolean logic on Fridays. (@richardjrossiii can confirm this).

It might be worth exploring the optimization where you don't need to call `unobserve:` twice, since the only piece that you need to guard is `removeObserver:` method call.

Here is the rough logic in my head around this optimization:

`-observe`:
After `addObserver:` call - if state is `.Initial` - then we change it to `.Observing`, otherwise we call `removeObserver:`.

`-unobserve`:
`.Observing` state - remove the object from the `infos` hash table + `removeObserver` is called inside `unobserve`.
`.Unobserved` state - remove the object from the `infos` hash table.